### ISSUE TITLE
'dz' counts a lot

### ISSUE THREAD

**Issue by FuncUny** [miniufo/xinvert#31]
Hi, 

I'm trying to use the method in a domain which has 19 levels in depth (totally 1000m), and I found it makes a poor inversion result. However,  it comes a much better result after I set my vertical levels as the data used in oceanic example (200 levels for 1000m), which convinced me that 'dz' counts a lot in calculation. But with the improvement comes a fat data array ( [360,200,1700,650] ) that is so hard to run forward. So I’d like to know, from your perspective, what value of 'dz' (or levels in depth) can allows me to reduce the size of data while maintaining a good accuracy?

Thanks a lot!

**Comment by miniufo** [comment]
Hi, are you using evenly distributed vertical grid? Currently xinvert only does his job on evenly distributed grid.

For large array, you need chunks to help you.  Which problem are you looking at?  Usually it is chunked along time.  But also depends on your specific case.

**Comment by FuncUny** [comment]
That make sence ! I've tried an evenly distributed vertical grid but with 20 levels (totally 1000m), and it comes a nice reslut. Thank you for the patience. BTW, is 'dz' important in the calculation ? Beacuse it seems that to keep an evenly distributed grid is more important, and both evenly distributed case (200 or 20 levels in totally 1000m) show nice accuracy.

**Comment by miniufo** [comment]
> BTW, is 'dz' important in the calculation ?

That's right, dx, dy, dz are all important in the calculations of derivatives along each direction.  If they are not evenly distributed, the results are not correct.  Right now it is not easy to add support to unevenly distributed grids.  But it is possible in a future release.

**Comment by navidcy** [comment]
If xinvert requires equally spaced vertical grid then perhaps an assert statement is in order and perhaps an error w some informative warning to the user if the assertion fails?

**Comment by miniufo** [comment]
Hi @navidcy, that's necessary.  I am thinking about it, and will fix this soon.

**Comment by navidcy** [comment]
oh I'm not sure...

yeah, `if ... then raise Exception('...')` would also do the job!

**Comment by miniufo** [comment]
This is done at [here](https://github.com/miniufo/xinvert/blob/9657ac140f54f133df8420394fd6390a66186fb4/xinvert/apps.py#L2175C1-L2175C39).  So if a coordinate is non-uniform, it will raise an error now.
